fecgov/openFEC
# Fix sort validation to skip 500 error

## Summary (required)
After this PR, when pass Invalid sort options, api will return 422, not 500 error code.
- Resolves #5588 
- #5626

### Required reviewers

1 dev

Completion criteria
- [x] Add sort option validation, when pass Invalid sort options, api will return 422, not 500 error code.
- [x] Add auto test
- [x] Update endpoint note on [google drive](https://docs.google.com/spreadsheets/d/1AkQjXA2XMQfxlRyinSY0uynhf2ZRMh7l_BTWdJgHzXY/edit#gid=408125426)

## Impacted areas of the application

General components of the application that this PR will affect:
/v1/candidates/totals/aggregates/
/v1/schedules/schedule_a/by_size/by_candidate/
/v1/schedules/schedule_a/by_state/by_candidate/
/v1/schedules/schedule_a/by_state/by_candidate/totals/
/v1/schedules/schedule_e/totals/by_candidate/
/v1/communication_costs/totals/by_candidate/
/v1/electioneering/totals/by_candidate/

<img width="550" alt="sort" src="https://github.com/fecgov/openFEC/assets/24395751/ff3137d7-b485-4971-a3b0-261fcc3bcf86">


## How to test
1) checkout branch
2) pytest
3) test urls, compare with prod api and make sure not return 500 error instead 422 error.

**[receipts](https://api.open.fec.gov/developers/#/receipts)**
**/v1/schedules/schedule_a/by_size/by_candidate/**
https://api.open.fec.gov/v1/schedules/schedule_a/by_size/by_candidate/?&sort=test&candidate_id=H8CA05035&cycle=2022&election_full=true&api_key=DEMO_KEY

http://127.0.0.1:5000/v1/schedules/schedule_a/by_size/by_candidate/?&sort=test&candidate_id=H8CA05035&cycle=2022&election_full=true

**/v1/schedules/schedule_a/by_state/by_candidate/**
https://api.open.fec.gov/v1/schedules/schedule_a/by_state/by_candidate/?sort=test&candidate_id=H8CA05035&cycle=2022&election_full=true&api_key=DEMO_KEY

http://127.0.0.1:5000/v1/schedules/schedule_a/by_state/by_candidate/?sort=test&candidate_id=H8CA05035&cycle=2022&election_full=true

**/v1/schedules/schedule_a/by_state/by_candidate/totals/**
https://api.open.fec.gov/v1/schedules/schedule_a/by_state/by_candidate/totals/?sort=test&candidate_id=H8CA05035&cycle=2022&election_full=true&api_key=DEMO_KEY

http://127.0.0.1:5000/v1/schedules/schedule_a/by_state/by_candidate/totals/?sort=test&candidate_id=H8CA05035&cycle=2022&election_full=true


**[independent expenditures](https://api.open.fec.gov/developers/#/independent%20expenditures)**
**/v1/schedules/schedule_e/totals/by_candidate/**
https://api.open.fec.gov/v1/schedules/schedule_e/totals/by_candidate/?election_full=true&sort=test&api_key=DEMO_KEY

http://127.0.0.1:5000/v1/schedules/schedule_e/totals/by_candidate/?election_full=true&sort=test


**[communication cost](https://api.open.fec.gov/developers/#/communication%20cost)**
**/v1/communication_costs/totals/by_candidate/**
https://api.open.fec.gov/v1/communication_costs/totals/by_candidate/?election_full=true&sort=test&api_key=DEMO_KEY

http://127.0.0.1:5000/v1/communication_costs/totals/by_candidate/?election_full=true&sort=test


**[electioneering](https://api.open.fec.gov/developers/#/electioneering)**
**/v1/electioneering/totals/by_candidate/**
https://api.open.fec.gov/v1/electioneering/totals/by_candidate/?election_full=true&sort=test&api_key=DEMO_KEY

http://127.0.0.1:5000/v1/electioneering/totals/by_candidate/?election_full=true&sort=test


**[candidate](https://api.open.fec.gov/developers/#/candidate)**
**/v1/candidates/totals/aggregates/**
https://api.open.fec.gov/v1/candidates/totals/aggregates/?aggregate_by=office-state-district&sort=test&election_full=true&api_key=DEMO_KEY

http://127.0.0.1:5000/v1/candidates/totals/aggregates/?aggregate_by=office-state-district&sort=test&election_full=true


test '/candidate/totals/' sort by candidate_id
http://127.0.0.1:5000/v1/candidates/totals/?cycle=2020&election_full=true&sort=candidate_id
